Doc/using/cmdline.rst

```diff
@@ -994,3 +994,117 @@ Debug-mode variables
    Need Python configured with the :option:`--with-trace-refs` build option.
 
    .. versionadded:: 3.11
+
+.. _using-on-finding-modules:
+
+Finding modules
+---------------
+
+A module search path is initialized when Python starts. This module search path
+may be accessed at :data:`sys.path`.
+
+The first entry in the module search path is the directory that contains the
+input script, if there is one. Otherwise, the first entry is the current
+directory, which is the case when executing the interactive shell, a ``-c``
+command, or ``-m`` module.
+
+The :envvar:`PYTHONPATH` environment variable is often used to add directories
+to the search path. If this environment variable is found then the contents are
+added to the module search path.
+
+.. note::
+
+   :envvar:`PYTHONPATH` will affect all installed Python versions/environments.
+   Be wary of setting this in your shell profile or global environment variables.
+   The :mod:`site` module offers more nuanced techniques as mentioned below.
+
+The next items added are the directories containing standard Python modules as
+well as any :term:`extension module`\s that these modules depend on. Extension
+modules are .dll files on Windows, .dylib files on macOS and .so files on Linux.
```

Comment:
Sorry, I may have mislead you here.  Extension modules have a `.so` extension on macOS as well as on Linux and on other Unix-y systems - all other ones AFAIK (like the BSD's). What is true is that shared libraries on macOS typically have a `.dylib` extension but Python extension modules on macOS aren't shared libraries. (A package installed (via pip, say) that includes an extension module might also install a shared library but I think that level of detail isn't needed here.) So perhaps this should be reworded to be `.dll files on Windows, .so files on other platforms`. ?

Reply:
Extension modules are ".pyd" files in Windows, which in CPython are DLL files (`IMAGE_FILE_DLL` to be specific in terms of the PE `IMAGE_FILE_HEADER.Characteristics`). The ".pyd" file extension is used in Windows in order to associate a Python icon with the file, as defined by the "Python.Extension" progid.

Reply:
No problem @ned-deily. Thanks @eryksun , I never knew why Windows used .pyd, always seemed curious to me that Python had these renamed .dll files!